NuRadioMC/simulation/simulation.py

```diff
@@ -1,85 +1,1032 @@
+                         time_logger=None,
+                         min_efield_amplitude=None,
+                         distance_cut=None):
+    """
+    Calculate the simulated electric field for a given shower and channel.
+
+    Parameters
+    ----------
+    showers : list of showers
+        A list of the showers that should be simulated (as defined in the input HDF5 file)
+    sid : int
+        the station id, only needed to identify the correct channel in the detector description
+    cid : int
+        the channel id for which the electric field should be calculated as defined in the detector description
+    det : Detector object
+        the detector description defining all channels. 
+    propagator : propagator object
+        the propagator that should be used to calculate the electric field from emitter to receiver, typically a ray tracer
+    medium: medium object
+        the medium in which the electric field is propagating, typically ice
+    config : dict
+        the NuRadioMC configuration dictionary (from the yaml file)
+    time_logger: time_logger object
+        the time logger to be used for the simulation
+    min_efield_amplitude: float
+        speedup cut: the minimum electric field amplitude, if all efields from all showers are belwo this threshold value
+        the station will not be set as candidate station.
+
+    Returns
+    -------
+    list of SimEfield objects
+        A list of SimEfield objects, one for each shower and propagation solution
+
+    """
+    if distance_cut is not None:
+        time_logger.start_time('distance cut')
+        vertex_positions = []
+        shower_energies = []
+        for i, shower in enumerate(showers):
+            vertex_positions.append(shower.get_parameter(shp.vertex))
+            shower_energies.append(shower.get_parameter(shp.energy))
+        vertex_positions = np.array(vertex_positions)
+        shower_energies = np.array(shower_energies)
+        vertex_distances = np.linalg.norm(vertex_positions - vertex_positions[0], axis=1)
+        time_logger.stop_time('distance cut')
+    logger.debug("Calculating electric field for station %d , channel %d from list of showers", sid, cid)
+    p = propagator # shorthand for more compact coding
+
+    sim_station = NuRadioReco.framework.sim_station.SimStation(sid)
+    sim_station.set_candidate(False)
+    if min_efield_amplitude is None:
+        sim_station.set_candidate(True)
+    sim_station.set_is_neutrino()  # naming not ideal, but this function defines in-ice emission (compared to in-air emission from air showers)
+
+    x2 = det.get_relative_position(sid, cid) + det.get_absolute_position(sid)
+    dt = 1. / config['sampling_rate']
+    # rescale the number of samples to the internal (higher) sampling rate used in the simulation
+    n_samples = det.get_number_of_samples(sid, cid) / det.get_sampling_frequency(sid, cid) / dt
+    n_samples = int(np.ceil(n_samples / 2.) * 2)  # round to nearest even integer
```

Comment:
[`numpy.rint`](https://numpy.org/doc/stable/reference/generated/numpy.rint.html#numpy-rint) can be used?